### PR TITLE
chore: upgrade to Angular 14.0.0

### PR DESCRIPTION
**Pull request by SamuelMarks**
Issues:
> DEPRECATED: The 'defaultProject' workspace option has been deprecated. The project to use will be determined from the current working directory.

…and:

> Package "@nguniversal/builders" has an incompatible peer dependency to "@angular-devkit/build-angular" (requires "^13.0.2", would install "14.0.0").
                  Package "@ngtools/webpack" has an incompatible peer dependency to "typescript" (requires "~4.4.3", would install "4.7.2").

---

Approach:

```sh
$ git clone --depth=1 https://github.com/angular/flex-layout
$ git remote add mine https://github.com/SamuelMarks/flex-layout
$ yarn
$ ng update --force $(jq -r '.dependencies * .devDependencies | keys | map(select(startswith("@angular"))) | @tsv' package.json)
$ git checkout -b ng14
$ git commit -S -am 'chore: upgrade to Angular 14.0.0'
$ git push --set-upstream mine ng14
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "flex-layout-srcs",
   "description": "Angular Flex Layout",
-  "version": "13.0.0-beta.38",
+  "version": "14.0.0-beta.38",
```
Comment (michaelfaith) [inline package.json:4]: They usually increment the beta version on new releases.  
```suggestion
  "version": "14.0.0-beta.39",
```

**Comment by PzYon** [comment]
Hi there. Any idea when we can expect the release of a new version? Would be really great so we can update to Angular 14. Thank you very much.

**Comment by CaerusKaru** [comment]
There are a couple of issues that I'd like to resolve before we cut a release. I have no issue if @SamuelMarks would like to remove the version bump for this, that way at least the nightly builds will support v14. But I want to close out at least #1396 and #1394, which were regressions of the last version. I've slated some time for this week to investigate these, and if all goes to plan, we could see a release sometime next week. In the meantime, there are some NPM and Yarn workarounds you can use to migrate.

Also big thanks to @SamuelMarks for taking on this work (this PR, not the issues above), it is much appreciated.

**Comment by damingerdai** [comment]
what time will the pr be merged?

**Comment by kuncevic** [comment]
@CaerusKaru Running `yarn` then after it fail I am running `yarn run postinstall --force` which is fixing local build for now
However this workaround does not work for github actions, any ideas how to please gihub actions?

**Comment by CaerusKaru** [comment]
Thinking the browserstack issues might be caused by permissions of some sort, I opened #1409 (with some other stuff I wanted to include), but that's failing too with the same issue. I'll reach out to the dev infra team to help resolve this.

**Comment by michaelfaith** [comment]
@CaerusKaru for this and future updates, would it be possible to use peer deps similar to how Angular Material does, which includes the next major version as well, to easy the update process.  e.g. `^14.0.0 || ^15.0.0` (https://github.com/angular/components/blob/5816a12dfa816e3f340627fdf561658de0e8c970/packages.bzl#L4)  It's not usually the case that the next major Angular or CDK versions break this library, so it's likely safe to do and would improve the DX for this interim period?

**Comment by JanneHarju** [comment]
I haven't use CircleCI so not sure how it should work but there is two Errors even there is Success in the end. Both errors say this "DEPRECATION: Passing custom equality testers to MatchersUtil#equals is deprecated." I point it out here if you missed it. https://app.circleci.com/pipelines/github/angular/flex-layout/270/workflows/10db3ad6-18ef-47d8-8dee-9c25b572db2a/jobs/7792

File: projects/libs/flex-layout/package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "tslib": "^2.3.0"
   },
   "peerDependencies": {
-    "@angular/cdk": "^13.0.0",
+    "@angular/cdk": "^14.0.0",
```
Comment (michaelfaith) [inline projects/libs/flex-layout/package.json:32]: Would it be possible to use peer deps similar to how Angular Material does, which includes the next major version as well, to ease the update process.  e.g. `^14.0.0 || ^15.0.0`.  This should be safe and would improve the DX for this interim period between angular and flex layout releases.

**Comment by kasir-barati** [comment]
any progress on this? 

```cmd
pnpm add @angular/flex-layout @angular/cdk
Already up-to-date
Progress: resolved 543, reused 523, downloaded 0, added 0, done
 WARN  Issues with peer dependencies found
.
└─┬ @angular/flex-layout
  ├── ✕ unmet peer @angular/cdk@^13.0.0: found 14.0.2
  ├── ✕ unmet peer @angular/core@^13.0.0: found 14.0.1
  ├── ✕ unmet peer @angular/common@^13.0.0: found 14.0.1
  └── ✕ unmet peer @angular/platform-browser@^13.0.0: found 14.0.1
```

It is really annoying and unstable to use `--force`

File: projects/libs/flex-layout/package.json (hunk 2 above)
Comment (michaelfaith) [inline projects/libs/flex-layout/package.json:32]: Pardon the repeated comment, I wanted to point out the specific line this was referring to

> @CaerusKaru for this and future updates, would it be possible to use peer deps similar to how Angular Material does, which includes the next major version as well, to ease the update process. e.g. `^14.0.0 || ^15.0.0` (https://github.com/angular/components/blob/5816a12dfa816e3f340627fdf561658de0e8c970/packages.bzl#L4) It's not usually the case that the next major Angular or CDK versions break this library, so it's likely safe to do and would improve the DX for this interim period between angular and flex layout releases.

File: projects/libs/flex-layout/package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "tslib": "^2.3.0"
   },
   "peerDependencies": {
-    "@angular/cdk": "^14.0.0",
+    "@angular/cdk": "^13.0.0 || ^14.0.0",
```
Comment (michaelfaith) [inline projects/libs/flex-layout/package.json:32]: @SamuelMarks it should go the other way. ^14.0.0 || ^15.0.0 that way when Angular 15 comes out, people with Flex-Layout 14 won't have trouble upgrading to angular 15, while this team works on updating flex-layout to 15

Comment (SamuelMarks) [reply projects/libs/flex-layout/package.json:32]: Ah, ok just did `fastmod -F '^13.0.0 || ^14.0.0' '^14.0.0 || ^15.0.0'`

**Pull request closed without merging**